Code/Tools/AssetProcessor/native/AssetDatabase/AssetDatabase.cpp

```diff
@@ -1117,10 +1120,18 @@ namespace AssetProcessor
             if (m_databaseConnection->ExecuteOneOffStatement(CREATE_STATS_TABLE))
             {
                 foundVersion = DatabaseVersion::AddedStatsTable;
-                AZ_TracePrintf(AssetProcessor::ConsoleChannel, "Upgraded Asset Database to version %i (AddedStatsTable)\n", foundVersion)
+                AZ_TracePrintf(AssetProcessor::ConsoleChannel, "Upgraded Asset Database to version %i (AddedStatsTable)\n", foundVersion);
             }
         }
 
+        if(foundVersion == DatabaseVersion::AddedStatsTable)
+        {
+            // Version update - change SourceDependency Source to SourceGuid column
+            // Do nothing so the whole database is dropped.
+            // Unfortunately we have to reprocess all assets because of the way the fingerprinting algorithm works,
+            // changing from storing the path to the UUID changes the fingerprint, resulting in all assets reprocessing anyway
```

Comment:
maybe add a log message? It just seems odd to submit comment out code with a conditional...